### DISCUSSION TITLE
Two way binding when using IObservables

### DISCUSSION THREAD

**Discussion by MaxMahem** [AvaloniaUI/Avalonia#16924] [Q&A]
So I've fallen in love with using IObservables to bind data in Avalonia. So much so that I've created a some little helper types to help me work it. For example:
```cs
public sealed class ObservableValue<T>(T initialValue) : IDisposable, IObservableValue<T> {
    readonly BehaviorSubject<T> valueSubject = new(initialValue);

    public T Value {
        get => this.valueSubject.Value;
        set => this.valueSubject.OnNext(value);
    }

    public IObservable<T> Changes => this.valueSubject.AsObservable();
}
```
Which works great for binding in one direction or the other.
```xml
<TextBlock Text="{Binding MyObservableValue.Changes^}" /><!-- update Text when Value changes -->
<TextBox Text="{Binding MyObservableValue.Value}"/><!-- report via Changes when user updates Text -->
```

So far, so great. However, sometimes, I want to do a two-way bind. And now, this approach falls short. I cannot bind to both the scalar value and the observable at the same time. It seems internally that Avalonia is using an IObserverable/IObserver pair for these two-way binds, which is great; I can easily modify my approach to accept that. However, I cannot figure out how to create such a binding.

Ideally, I would be able to create this via a markup extension, and just expose my two-way binding object that could get transformed into a binding (perhaps by `Binding.Instance`?). Something like:
```xml
<TextBox Text="{myMu:Subject {Binding MyObservableValue}}" />
```
Then my markup does something like:
```cs
public class SubjectMarkupExtension(BindingBase observableValue) {
   public Binding ProvideValue(IServiceProvider serviceProvider) {
       // build the binding somehow?
    }
}
```
But I'm at a loss for how this might be accomplished. Or even if this is the right approach. Any insight would be appreciated!

**Comment by fitdev** [comment]
I would very much want the same behavior, but have not figured out a way to do it either.
I asked something similar here: https://github.com/AvaloniaUI/Avalonia/discussions/16257
As a result of which there was a related issue created that tracks this: #16260

**Comment by JaggerJo** [comment]
We're also using Observables everywhere for everything. 

Our pattern looks ruffly like what @MaxMahem lined out above. Would be great to have first class support for pushing values to observers.
